packages/core/src/Account.ts

```diff
@@ -351,69 +341,45 @@ export class Account extends TypedEventEmitter<Events> {
     const client = await this.service.client.register(loginData, clientInfo, initialPreKeys);
     const clientId = client.id;
 
-    if (this.service.mls) {
-      const {userId, domain = ''} = this.apiClient.context;
-      await this.initMLSClient({id: userId, domain}, client);
-    }
-    this.logger.info(`Created new client {mls: ${!!this.service.mls}, id: ${clientId}}`);
-
     await this.service.notification.initializeNotificationStream(clientId);
     await this.service.client.synchronizeClients(clientId);
-
-    return this.initClient(client);
```

Comment:
No more hidden agenda of the `registerClient` method. It now really only registers the client on backend (the cryptographic material has to be initialized by the consumer)